addons/io_hubs_addon/components/definitions/loop_animation.py

```diff
@@ -1,9 +1,160 @@
 import atexit
 import bpy
-from bpy.props import StringProperty, CollectionProperty, IntProperty, BoolProperty
+from bpy.app.handlers import persistent
+from bpy.props import StringProperty, CollectionProperty, IntProperty, BoolProperty, EnumProperty
 from bpy.types import PropertyGroup, Menu, Operator
 from ..hubs_component import HubsComponent
 from ..types import Category, PanelType, NodeType
+from ..utils import redraw_component_ui
+
+nla_track_name_msgbus_owner = None
+action_name_msgbus_owner = None
+
+class TrackPropertyType(PropertyGroup):
+    name: StringProperty(
+        name="Display Name",
+        description="Display Name",
+    )
+    track_name: StringProperty(
+        name="Track Name",
+        description="Track Name",
+    )
+    action_name: StringProperty( # Will only contain data if the track name is generic
+        name="Action Name",
+        description="Action Name",
+    )
+    track_type: EnumProperty(
+        name="Track Type",
+        description="Track Type",
+        items=[
+            ("object", "Object", "Object"),
+            ("shape_key", "Shape Key", "Shape Key")
+            ],
+        default="object"
+    )
+
+bpy.utils.register_class(TrackPropertyType)
+
+@atexit.register
+def unregister():
+    bpy.utils.unregister_class(TrackPropertyType)
```

Comment:
I've landed a PR that changes how we handle module-wide class registration. It's a pretty simple fix, you just need to move this to a module `register_module`/`unregister_module` functions.